src/library.js

```diff
@@ -4677,6 +4679,18 @@ LibraryManager.library = {
     }
   },
 
+  emscripten_get_module_name: function(buf, length) {
+    return stringToUTF8(wasmBinaryFile, buf, length);
```

Comment:
Sorry for the late review!

`wasmBinaryFile` will not exist in wasm2js mode, or if the wasm was embedded in the JS. But maybe it's not worth trying to support more cases than the common one.

Reply:
It appears to exist in wasm2js mode, which is why I did this:

```console
$ emcc -fsanitize=null null-read.c -s WASM=0
emcc:WARNING: emcc: JS support in the upstream LLVM+wasm2js path is very experimental currently (best to use fastcomp for asm.js for now)
emscripten:WARNING: Wasm source map won't be usable in a browser without --source-map-base
$ grep wasmBinaryFile a.out.js 
var wasmBinaryFile = 'a.out.wasm';
if (!isDataURI(wasmBinaryFile)) {
  wasmBinaryFile = locateFile(wasmBinaryFile);
      return Module['readBinary'](wasmBinaryFile);
    return fetch(wasmBinaryFile, { credentials: 'same-origin' }).then(function(response) {
        throw "failed to load wasm binary file at '" + wasmBinaryFile + "'";
        !isDataURI(wasmBinaryFile) &&
      return WebAssembly.instantiateStreaming(fetch(wasmBinaryFile, { credentials: 'same-origin' }), info)
      return stringToUTF8(wasmBinaryFile, buf, length);
$ node a.out.js 
null-read.c:3:9: runtime error: load of null pointer of type 'int'
```

Reply:
I don't understand? The file doesn't exist, but the variable does, yes. Is just the variable name enough?

Reply:
Yes, it just wants a name.